### ISSUE TITLE
Router changes url link but not the view when using react router dom

### ISSUE THREAD

**Issue by r-ush** [Nazeh-Taha/react-responsive-animate-navbar#4]

**Comment by Nazeh-Taha** [comment]
I think this article will help you.
https://codeburst.io/getting-started-with-react-router-5c978f70df91

**Comment by ILoomans** [comment]
I love the design for the navbar but I am having the same issue, i had a working navigation bar, once i replaced it with this one i had the same issue that the url link changes but the view doesn't.

**Comment by r-ush** [comment]
yep same

**Comment by Nazeh-Taha** [comment]
Thanks for contacting me. I want to ask if you used "react-router-dom" in your project.

**Comment by r-ush** [comment]
yes, this package https://www.npmjs.com/package/react-router-dom

**Comment by Nazeh-Taha** [comment]
Can I see the repo or the code

**Comment by ZachPfeifer** [comment]
Did this issue get resolved? I have the same problem where it will change the URL but not the view. My original Router Links work just fine but the navbar isn't working. I'll link my repo but you won't be able to clone and host without my keys.

Repo:
-https://github.com/ZachPfeifer/silver-spirit-alchemy/blob/master/client/src/components/NavBar/Navbar4.js

**Comment by Nazeh-Taha** [comment]
> Did this issue get resolved? I have the same problem where it will change the URL but not the view. My original Router Links work just fine but the navbar isn't working. I'll link my repo but you won't be able to clone and host without my keys.
> 
> Repo:
> -https://github.com/ZachPfeifer/silver-spirit-alchemy/blob/master/client/src/components/NavBar/Navbar4.js

ok try this:
import the react-router like this:
import { BrowserRouter as Router, Route } from "react-router-dom";
then but this:
   <Router>
      <Route path="/" component={Home} exact />
      <Route path="/store" component={Store} />
      <Route path="/about" component={AboutUs} />
......
    </Router>

and remove all (exact) from all routes except home route.

**Comment by Nazeh-Taha** [comment]
> > Did this issue get resolved? I have the same problem where it will change the URL but not the view. My original Router Links work just fine but the navbar isn't working. I'll link my repo but you won't be able to clone and host without my keys.
> > Repo:
> > -https://github.com/ZachPfeifer/silver-spirit-alchemy/blob/master/client/src/components/NavBar/Navbar4.js
> 
> ok try this:
> import the react-router like this:
> import { BrowserRouter as Router, Route } from "react-router-dom";
> then but this:
> 
> 
> 
> 
> ......
> and remove all (exact) from all routes except home route.

```
    <Router>
      <Route path="/" component={Home} exact />
      <Route path="/about" component={AboutUs} />
      <Route path="/cart" component={Cart}/>
    </Router>
```

**Comment by ZachPfeifer** [comment]
The issue persists and I only have problems with the navbar. I have buttons that work just fine using the react-router-dom <Link>. 

Here is the condensed version of my App.js:

import React from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";

 <Router>
        <Route exact path="/" component={Home} />
        <Route path="/store" component={Store} />
        <Route path="/store/:payload" component={SingleProduct} />
        <Route path="/about" component={AboutUs} />
        <Route path="/cart" component={Cart} />
        <Route component={Error} />
      </Router>

**Comment by JevF1982** [comment]
Very nice navbar , but I have indeed the same problem with react-router...

**Comment by Nazeh-Taha** [comment]
Hi guys, sorry if there is a problem in the navbar just give me two days and I will fix it.
thanks a lot for using this package and thanks for contacting me.
I will tell you when I finish fixing it.

**Comment by umar834** [comment]
Loved the design but facing the same issue.. :(

**Comment by user32-2** [comment]
I created a PR that resolved this issue for me on mobile devices.

**Comment by hernandez87v** [comment]
Really nice Navbar but having the same issue with react-router-dom
` <Router>
      <div className="App">
        <Navbar />
        <Route path="/" exact component={Twitch} />
        <Route path="/Twitch" exact component={Twitch} />
        <Route path="/Soundcloud" exact component={Soundcloud} />
        <Route path="/Mixcloud" exact component={Mixcloud} />
      </div>
    </Router>`

On a side note anyway to just have text instead of a logo?

**Comment by dawood6** [comment]
Same Issue Again Any Help

**Comment by dawood6** [comment]
> Hi guys, sorry if there is a problem in the navbar just give me two days and I will fix it.
> thanks a lot for using this package and thanks for contacting me.
> I will tell you when I finish fixing it.

bro I like the design but not using it because of navigation issue

**Comment by danielro8** [comment]
Yes, I have the same problem.
A pity because it's an awesome design.

**Comment by christorepl** [comment]
the pull request didn't seem to change anything, neither did the advice given by the owner. such a shame, it's a really simple navbar to setup. i hope to see it fixed soon!

**Comment by mohanor** [comment]
same

**Comment by mohanor** [comment]
i have the same problem

**Comment by lorstenoplo** [comment]
I have fixed the page change bug in PR : #21

**Comment by pr7prashant** [comment]
> I have fixed the page change bug in PR : #21

@Nazeh-Taha are you planning to publish this to the NPM registry soon ?

**Comment by lorstenoplo** [comment]
@pr7prashant 
It should not happen yet I guess, because there are some small css positioning issues of the 
navbar and gsap related issues

**Comment by pr7prashant** [comment]
Thanks @lorstenoplo 
I also noticed that the navbar is not sticky on top for mobile devices. Can I create a feature request for this ?

**Comment by lorstenoplo** [comment]
@pr7prashant  yeah you can

**Comment by karlaluuu** [comment]
has anyone made it work? i added the component, as well as the route under. still not working

**Comment by lorstenoplo** [comment]
Did you install the package using mom or yarn

**Comment by karlaluuu** [comment]
@lorstenoplo did use npm. i'll try reinstalling the package and see

**Comment by lorstenoplo** [comment]
**Don't** there is no use because he ( @Nazeh-Taha  ) hasn't published the latest version to the npm registry.

You can either wait for the latest release or
Clone this repository instead ( #23 actually ) then :
1. Run `npm install` in the right directory
2. Run `npm run build`
3.  Copy the `dist` folder generated by the build command 
4.  Now go to the directory where your project is located
5.  Navigate to `node_modules > react-responsive-animate-navbar` then  replace the existing `dist` folder with the newly generated one

**Comment by karlaluuu** [comment]
hey, sorry for the late response. thank you, @lorstenoplo! made it work! 

anyways, very good design as well. kudos!!!!

**Comment by karlaluuu** [comment]
> @karlaluuu **Don't** there is no use because he ( @Nazeh-Taha ) hasn't published the latest version to the npm registry.
> 
> You can either wait for the latest release or
> Clone this repository instead ( #23 actually ) then :
> 
> 1. Run `npm install` in the right directory
> 2. Run `npm run build`
> 3. Copy the `dist` folder generated by the build command
> 4. Now go to the directory where your project is located
> 5. Navigate to `node_modules > react-responsive-animate-navbar` then  replace the existing `dist` folder with the newly generated one

question. is there any timeline to get the latest version out in the npm registry? just realized that this method works locally, but not being catched by my site deployed in a web hosting company. anyone using netlify to host their site and made it work? :(
